src/BusinessLayer/InternetReporting.cpp

```diff
@@ -0,0 +1,494 @@
+#include <cstring>
+
+#include <QIODevice>
+#include <QDateTime>
+#include <QJsonArray>
+#include <QString>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "CcsDefines.h"
+#include "InternetReporting.h"
+#include "BatteryData.h"
+#include "BatteryFaultsData.h"
+#include "CmuData.h"
+#include "DriverControlsData.h"
+#include "KeyMotorData.h"
+#include "LightsData.h"
+#include "MotorDetailsData.h"
+#include "MotorFaultsData.h"
+#include "MpptData.h"
+#include "CommunicationService.h"
+#include "View.h"
+
+QJsonArray KeyMotor;
+QJsonObject MotorDetails0;
+QJsonObject MotorDetails1;
+QJsonObject DriverControls;
+QJsonArray MotorFaults;
+QJsonObject BatteryFaults;
+QJsonObject Battery;
+QJsonArray CMUArray;
+QJsonArray mPPTArray;
+QJsonObject lightsInfo;

```

Comment:
This data really doesn't need to shared as global variables. There are very very very few uses cases, that you need a global variable. Also in general, CapCase is reserved for types, if a word doesn't define a type, it shouldn't be CapCase.

Global variables are:
1. Confusing to read because they're hard to look for
2. Not testable
3. Not extensible

A much better alternative to your make functions is to return the created object rather than use a shared global variable.